### PR TITLE
TIMM support in the application

### LINKED ISSUE #7100: Backend: Support timm catalog integration and dynamic manifest resolution
Integrate the offline `timm` snapshot catalog from #7095 into the backend service to enable dynamic model selection:
- **Manifest Resolution:** Implement `TimmManifestProvider` to lazily construct `ModelManifest` instances on demand from `timm_catalog_snapshot.json` while serving a single synthetic timm card entry in the standard classification architecture listing.
- **Catalog Search API:** Expose a paginated `GET /api/model_architectures/timm/backbones` endpoint supporting search, limit/offset pagination, and family facet filtering.
- **Schema Updates:** Extend backend domain models (`ModelManifest`) and DTOs to support `family`, `version`, and optional `timm_managed` weight sources.
- **Config Conversion:** Update `GetiConfigConverter` to route `image-classification-timm-*` IDs to `timm_generic.yaml` and inject arch-specific data_input_params (`input_size`, `mean`, `std`).
- **Hyperparameter Handling:** Adapt `HyperparametersUpdater` to write the learning rate directly to model initialization arguments when no optimizer block is present in the recipe.

Refer to [design doc#section 4](https://github.com/open-edge-platform/geti/blob/develop/library/docs/design/timm-integration.md#4-required-backend-changes) for more details

### PR DESCRIPTION
**Pull request by itallix**
### Summary

- [x] New APIs to retrieve `family`/`model`/`tag` from the catalog
- [x] Added dynamic manifest builder
- [x] Removed `std`/`mean`/`interpolation` from the catalog json
- [x] Refactored `GetiConfigConverter`: extracted `RecipeResolver`

### How to test

Run training via `curl` with any model from the catalog json:

```
curl -k -X POST -H "Content-Type: application/json" https://localhost:7860/api/jobs -d '{"job_type": "train", "project_id": "$PROJECT_ID", "parameters": { "model_architecture_id": "image-classification-timm-efficientnet_b1_pruned.in1k", "device": "cpu" }}' | jq
```

### Checklist

- [x] The PR title and description are clear and descriptive
- [x] I have manually tested the changes
- [ ] All changes are covered by automated tests
- [x] All related issues are linked to this PR (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

File: application/backend/app/models/model_manifest.py [hunk 1]
```diff
@@ -115,6 +111,13 @@ def local_filename(self) -> str:
         return self.cache_filename or Path(self.url).name
 
 
+class WeightsSource(StrEnum):
+    """Source of pretrained weights for a model architecture."""
+
+    TIMM_MANAGED = "timm_managed"  # Weights are managed by the timm library
+    GETI_HOSTED = "geti_hosted"  # Weights URL and checksum are defined directly in the manifest file
```
Comment (leoll2) [inline application/backend/app/models/model_manifest.py]: "Geti hosted" may be misleading, suggesting that the weights are hosted in a repository controlled by our organization.

I suggest to rename the values:
- `GETI_HOSTED` -> `DIRECT_LINK`
- `TIMM_MANAGED` -> `TIMM`

In the future, we may add another type of source to bypass the legal complications of weights mirroring: write the URLs to an external file (not committed to the repo) and download that file at runtime. If we decide to commit to that approach, it could be represented by another `WeightsSource` value, e.g. `WEIGHTS_MANIFEST`.

File: application/backend/app/models/model_manifest.py [hunk 2]
```diff
@@ -124,8 +127,27 @@ class ModelManifest(BaseModel):
...
+    )
+    pretrained: str | None = Field(
+        default=None,
+        title="Pretrained Tag",
+        description="Pretrained tag of the model (e.g. timm's pretraining tag such as 'in22k_ft_in22k')",
+    )
+    weights_source: WeightsSource = Field(
+        title="Weights Source",
+        description="Source of pretrained weights for the model architecture (timm-managed or geti-hosted)",
+        default=WeightsSource.GETI_HOSTED,
```
Comment (leoll2) [inline application/backend/app/models/model_manifest.py]: I suggest to restructure the manifest a little to avoid so many new attributes at the root level:
- Group all weights-related attributes under `pretrained_weights`
- Group TIMM-specific metadata under `timm_metadata`

Example of Geti model:
```yaml
id: image-classification-vit-tiny
name: ViT-Tiny
pretrained_weights:
    source: direct_link
    url: https://storage.googleapis.com/abc/123
    mirror_url: https://mirror.storage.googleapis.com/abc/123
    sha: 0c5c4818e
```

Example of timm model:
```yaml
id: image-classification-timm-beit_base_patch16_224.in22k_ft_in22k
name: beit_base_patch16_224.in22k_ft_in22k
pretrained_weights:
    source: timm
timm_metadata:
    family: beit
    version: base_patch16_224
    checkpoint: in22k_ft_in22k
```

As for the Pydantic models, you could have two subclasses of `PretrainedWeights` - `TimmPretrainedWeights` and `DirectLinkPretrainedWeights` - discriminated by `source`. This would also make the validation stronger, because all attributes under `timm_metadata` can be made non-optional and you can simply validate that `timm_metadata is not None` when `pretrained_weights.source is WeightsSource.TIMM`.

File: application/backend/app/execution/common/geti_config_converter.py [hunk 3]
```diff
@@ -681,429 +674,106 @@ def convert(config: dict) -> dict:
...
+            "input_size": preprocessing["input_size"],
+        }
 
     @staticmethod
-    def _is_ultralytics_recipe(recipe_path: Path) -> bool:
-        """Return whether a recipe declares the Ultralytics backend."""
-        with recipe_path.open() as f:
-            recipe = yaml.safe_load(f)
-        return isinstance(recipe, dict) and recipe.get("backend") == "ultralytics"
+    def _assert_no_dynamic_placeholders(config: dict) -> None:
```
[the remark this answers is not in the document]
Comment (itallix) [reply application/backend/app/execution/common/geti_config_converter.py:763]: This is intentional: for numeric types - `nan` for strings - `DYNAMIC`

File: application/backend/app/api/schemas/model_architecture.py [hunk 4]
```diff
@@ -15,11 +15,14 @@ class ModelArchitectureView(BaseModel):
...
     description: str = Field(title="Description", description="Detailed description of the model capabilities")
+    timm_metadata: TimmMetadata | None = Field(
+        default=None, title="TIMM Model Metadata", description="Metadata about the specific TIMM model architecture"
+    )
     capabilities: Capabilities = Field(
         title="Model Capabilities", description="Special capabilities supported by the model"
     )
-    license: str = Field(title="License", description="License under which the model architecture is released")
-    stats: ModelStats = Field(title="Model Statistics", description="Statistics about the model")
+    license: str | None = Field(title="License", description="License under which the model architecture is released")
```
Comment (leoll2) [inline application/backend/app/api/schemas/model_architecture.py]: `license` should never be `None`. Models with unknown licenses should be excluded to prevent problems.

File: application/backend/app/api/routers/model_architectures.py [hunk 5]
```diff
@@ -35,11 +37,79 @@ def get_model_architectures(task: TaskType) -> ModelArchitectures:
     model_manifests = ModelManifestService.get_model_manifests()
     top_picks = RECOMMENDED_MODEL_ARCHITECTURES.get(task, None)
 
+    architectures = [
+        ModelArchitectureView.model_validate(manifest, from_attributes=True)
+        for manifest in model_manifests.values()
+        if manifest.task == task
+    ]
+    if task == TaskType.CLASSIFICATION:
+        architectures.append(_build_timm_card_entry())
```
Comment (leoll2) [inline application/backend/app/api/routers/model_architectures.py:46]: What happens when the UI feature flag is disabled? Does the UI ignore the new timm manifest returned by the endpoint?

https://github.com/open-edge-platform/geti/blob/a1211f44a20eafd64d51dd1ae73f6d5e86c53aba/application/ui/src/constants/feature-flags.ts#L6

File: application/backend/app/api/routers/model_architectures.py [hunk 6]
```diff
@@ -35,11 +37,79 @@ def get_model_architectures(task: TaskType) -> ModelArchitectures:
...
+    This is a fixed, non-trainable entry point rendered by the UI to open the
+    searchable timm backbone selector. It is intentionally never registered in
+    `ModelManifestService.get_model_manifests()` and its id is never resolvable
+    via `get_model_manifest_by_id` — selecting a concrete backbone always uses
+    a real `image-classification-timm-<name>` id instead.
+    """
+    return ModelArchitectureView(
+        id=TIMM_CARD_ID,
+        task=TaskType.CLASSIFICATION,
+        name="Custom backbone (timm)",
```
Comment (leoll2) [inline application/backend/app/api/routers/model_architectures.py]: I'd suggest to rename the card to "Pytorch Image Models (timm)"

File: application/backend/app/api/routers/model_architectures.py [hunk 7]
```diff
@@ -35,11 +37,79 @@ def get_model_architectures(task: TaskType) -> ModelArchitectures:
...
+        timm_metadata=None,
+        description="Choose any of 1700+ timm backbones.",
```
Comment (leoll2) [inline application/backend/app/api/routers/model_architectures.py]: If I understand correctly, we are not exposing the full collection of 1700 models, only those with an open-source "friendly" license. Please update the number to reflect our catalog size.

File: application/backend/app/execution/common/recipe_resolver.py
Comment (leoll2) [inline application/backend/app/execution/common/recipe_resolver.py:1]: Thanks for refactoring the converter, it's more readable now.

File: application/backend/app/api/schemas/model_architecture.py (hunk 4 above)
Comment (itallix) [reply application/backend/app/api/schemas/model_architecture.py]: We cannot infer the license for synthetic card type which is known only after additional attributes are specified (family, model, pretrained-tag). 
Would it make sense to provide some placeholder instead?

File: application/backend/app/api/routers/model_architectures.py (hunk 5 above)
Comment (itallix) [reply application/backend/app/api/routers/model_architectures.py:46]: cc @jpggvilaca

Comment (jpggvilaca) [reply application/backend/app/api/routers/model_architectures.py:46]: No, it just hides the TIMM configuration part (the new UI above "advanced settings")

File: application/backend/app/api/routers/model_architectures.py (hunk 7 above)
Comment (itallix) [reply application/backend/app/api/routers/model_architectures.py]: Indeed. I replaced the hardcoded value with the catalog length. For now it includes all models, but we will review it in #7395

File: application/backend/app/api/routers/model_architectures.py (hunk 5 above)
Comment (leoll2) [reply application/backend/app/api/routers/model_architectures.py:46]: Ok. Hopefully the timm feature will be ready for 3.2, but if it doesn't, then we shouldn't forget to hide/remove the card.

File: application/backend/app/api/routers/model_architectures.py [hunk 8]
```diff
@@ -35,11 +37,79 @@ def get_model_architectures(task: TaskType) -> ModelArchitectures:
...
+    `ModelManifestService.get_model_manifests()` and its id is never resolvable
+    via `get_model_manifest_by_id` — selecting a concrete backbone always uses
+    a real `image-classification-timm-<name>` id instead.
+    """
+    return ModelArchitectureView(
+        id=TIMM_CARD_ID,
+        task=TaskType.CLASSIFICATION,
+        name="Pytorch Image Models (timm)",
+        timm_metadata=None,
+        description=f"Choose any of {TimmCatalog.count_backbones()} timm backbones.",
```
Comment (leoll2) [inline application/backend/app/api/routers/model_architectures.py]: The description could be extended a bit:

> PyTorch Image Models (TIMM) is a large collection of SOTA image classification models and pretrained weights. Geti offers {TimmCatalog.count_backbones()} of these models across dozens of architecture families.

**Review by leoll2 — DISMISSED** [review]

File: application/backend/app/api/schemas/model_architecture.py (hunk 4 above)
Comment (itallix) [reply application/backend/app/api/schemas/model_architecture.py]: replaced with placeholder `varies by model`

**Review by yvolokitin — DISMISSED** [review]

File: application/backend/app/supported_models/timm/manifest_provider.py [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+    ModelManifest,
+    ModelManifestDeprecationStatus,
+    ModelStats,
+    TimmMetadata,
+    TimmPretrainedWeights,
+)
+from app.models.training_configuration import AlgoLevelParameters, AlgoLevelTrainingParameters
+from app.supported_models.timm.catalog import _snapshot
+
+_DEFAULT_LICENSE = "apache-2.0"
```
Comment (AlexanderBarabanov) [inline application/backend/app/supported_models/timm/manifest_provider.py]: I think this might mislead users. If there is a network failure or the model card does not contain a license field, the application will return Apache. Could we use `unknown` instead?

Comment (itallix) [reply application/backend/app/supported_models/timm/manifest_provider.py]: Added `unknown` with `TODO` to get rid of the network call (HF card sometimes misses the license information)

**Review by AlexanderBarabanov — APPROVED** [review]

**Review by leoll2 — APPROVED** [review]

**Review by yvolokitin — APPROVED** [review]
